packages/vaex-core/vaex/dataframe.py

```diff
@@ -5868,153 +5872,163 @@ def mangle_name(prefix, name, suffix):
         :param progress: progress callback that gets a progress fraction as argument and should return True to continue,
                 or a default progress bar when progress=True
-        :param: bool virtual: When True, export virtual columns
-        :param str sort: expression used for sorting the output
-        :param bool ascending: sort ascending (True) or descending
         :return:
         """
-        import vaex.arrow.export
-        vaex.arrow.export.export(self, path, column_names, byteorder, shuffle, selection, progress=progress, virtual=virtual, sort=sort, ascending=ascending)
+        self.export_arrow_stream(path, progress=progress, chunk_size=chunk_size, parallel=parallel)
 
-    def export_arrow_stream(self, path_or_writer, progress=None, chunk_size=default_chunk_size):
-        """Exports the DataFrame as Arrow stream
+    def export_arrow_stream(self, path_or_writer, progress=None, chunk_size=default_chunk_size, parallel=True):
+        """Exports the DataFrame as Arrow stream to a file or arrow writer
 
         :param path_or_writer path: path for file or :py:data:`pyarrow.RecordBatchStreamWriter`
         :param progress: progress callback that gets a progress fraction as argument and should return True to continue,
                 or a default progress bar when progress=True
+        :param int chunk_size: Number of rows for each table to write
+        :param bool parallel: {evaluate_parallel}
         :return:
         """
-        schema = self[0:1].to_arrow_table().schema
+        schema = self[0:1].to_arrow_table(parallel=False).schema
         progressbar = vaex.utils.progressbars(progress)
         def write(writer):
             progressbar(0)
             N = len(self)
-            for i1, i2, table in self.to_arrow_table(chunk_size=chunk_size):
+            if chunk_size:
+                for i1, i2, table in self.to_arrow_table(chunk_size=chunk_size, parallel=parallel):
+                    writer.write_table(table)
+                    progressbar(i2/N)
+                progressbar(1.)
+            else:
+                table = self.to_arrow_table(chunk_size=chunk_size, parallel=parallel)
                 writer.write_table(table)
-                progressbar(i2/N)
-            progressbar(1.)
         if isinstance(path_or_writer, str):
             with pa.OSFile(path_or_writer, 'wb') as sink:
                 writer = pa.RecordBatchStreamWriter(sink, schema)
                 write(writer)
         else:
             write(path_or_writer)
 
-    def export_parquet(self, path, column_names=None, byteorder="=", shuffle=False, selection=False, progress=None, virtual=True, sort=None, ascending=True):
-        """Exports the DataFrame to a parquet file
+    @docsubst
+    def export_parquet(self, path, progress=None, parallel=True, **kwargs):
+        """Exports the DataFrame to a parquet file.
+
+        Note: This may require that all of the data fits into memory (mmapped data is an exception).
```

Comment:
I would prefer if this is spelled out (i.e. memory mapped). Also not quite clear when the exception happens. I tried to export 60GB hdf5 file to parquet with this method, and i could not do it. Might be other issue (see general comments).